### PR TITLE
BaseExtendedPicker: Create contextmenu for renderedItem, fix auto focus

### PR DESCRIPTION
**Pull request by amyngu**
#### Pull request checklist

- [x] Include a change request file using `$ npm run change`

#### Description of changes

- Create component to wrap the rendered item, so users get contextual menu if certain props are present
- Get rid of loading state (causes flashing when using async onResolveSuggestions
- Fix auto focus on input afters suggestion selection
- Change example to use async calls, to better reflect app environment

### CODE REVIEW

**Comment by amyngu** [comment]
@dzearing @joschect ping :) I'd appreciate a review!

File: packages/experiments/src/components/SelectedItemsList/SelectedPeopleList/SelectedPeopleList.tsx [hunk 1]
```diff
@@ -93,10 +93,28 @@ export class SelectedPeopleList extends BasePeopleSelectedItemsList {
...
+              item={ props.item }
+            />
+          )
+          : renderedItem
+      );
     }
   }
 
+  @autobind
+  private _beginEditing(item: IExtendedPersonaProps): void {
```
Comment (joschect) [inline packages/experiments/src/components/SelectedItemsList/SelectedPeopleList/SelectedPeopleList.tsx]: Why use begin and complete editing? You should be able to pass a dismiss function to contextualmenu that will handle some of this.

File: packages/experiments/src/components/SelectedItemsList/SelectedPeopleList/Items/SelectedItemWithContextMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+export interface ISelectedItemWithContextMenuProps extends IBaseProps {
+  renderedItem: JSX.Element;
+  beginEditing?: (item: IExtendedPersonaProps) => void;
+  menuItems: IContextualMenuItem[];
+  item: IExtendedPersonaProps;
+}
+
+export class SelectedItemWithContextMenu extends BaseComponent<ISelectedItemWithContextMenuProps, IPeoplePickerItemState> {
```
Comment (joschect) [inline packages/experiments/src/components/SelectedItemsList/SelectedPeopleList/Items/SelectedItemWithContextMenu.tsx]: This almost feels too specific, could you style a button with contextualmenuitems to meet your needs?

**Comment by dzearing** [comment]
There's a weird conflict, can you fix @amyngu ?

File: packages/experiments/src/components/SelectedItemsList/SelectedPeopleList/SelectedPeopleList.tsx (hunk 1 above)
Comment (amyngu) [reply packages/experiments/src/components/SelectedItemsList/SelectedPeopleList/SelectedPeopleList.tsx]: The contextual menu will not be open throughout the duration of this, so I don't think using this dismiss function is correct here.

File: packages/experiments/src/components/SelectedItemsList/SelectedPeopleList/Items/SelectedItemWithContextMenu.tsx (hunk 2 above)
Comment (amyngu) [reply packages/experiments/src/components/SelectedItemsList/SelectedPeopleList/Items/SelectedItemWithContextMenu.tsx]: You're right. I will revisit this and try and create a better abstraction to fit my needs

**Pull request merged**